From 65dd3134fc76e389933aaa98700278f0fb287858 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 15:30:17 -0400
Subject: [block] mtip32xx: blk_mq_init_queue() returns an ERR_PTR

Message-id: <1402673421-17117-15-git-send-email-snitzer@redhat.com>
Patchwork-id: 83924
O-Subject: [RHEL7.1 PATCH 14/18] mtip32xx: blk_mq_init_queue() returns an ERR_PTR
Bugzilla: 1105703
RH-Acked-by: David Milburn <dmilburn@redhat.com>

BZ: 1105703

Upstream commit a8a642ccd2e86248ae23c5d762dac67eaea02423
Author: Dan Carpenter <dan.carpenter@oracle.com>
Date:   Wed May 14 15:54:18 2014 +0300

    mtip32xx: blk_mq_init_queue() returns an ERR_PTR

    We changed this from blk_alloc_queue_node() to blk_mq_init_queue() so
    the check needs to be updated as well.

    Fixes: ffc771b3ca8b2 ('mtip32xx: convert to use blk-mq')
    Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/block/mtip32xx/mtip32xx.c b/drivers/block/mtip32xx/mtip32xx.c
index f96fe0f..948b284 100644
--- a/drivers/block/mtip32xx/mtip32xx.c
+++ b/drivers/block/mtip32xx/mtip32xx.c
@@ -3886,7 +3886,7 @@ skip_create_disk:
 
  /* Allocate the request queue. */
  dd->queue = blk_mq_init_queue(&dd->tags);
- if (dd->queue == NULL) {
+ if (IS_ERR(dd->queue)) {
   dev_err(&dd->pdev->dev,
    "Unable to allocate request queue\n");
   rv = -ENOMEM;
-- 
1.7.1